### DISCUSSION TITLE
ContextShift vs Smart Context

### DISCUSSION THREAD

**Discussion by VL4DST3R** [LostRuins/koboldcpp#658] [Q&A]
There is a bit inside the [documentation ](https://github.com/LostRuins/koboldcpp/wiki#what-is-contextshift) about ContextShift that I'm not clear about: 

> So long as you use no memory/fixed memory and don't use world info, you should be able to avoid almost all reprocessing between consecutive generations even at max context. 

But what if you DO use one or both of those things? What happens then and how does it compare to using Smart Context in that (heh) context? Is it still preferred to use the new ContextShift regardless?

**Comment by LostRuins** [comment]
ContextShift is always better than Smart Context on models that support it.

**Reply by VL4DST3R** [reply]
And for older (GGML) models does it switch to using Smart Context or it must be manually specified by the user at startup?

**Reply by LostRuins** [reply]
No, both options must be manually configured.
